pkg/client/unversioned/clientcmd/merged_client_builder.go

```diff
@@ -115,10 +115,10 @@ func (config *DeferredLoadingClientConfig) ClientConfig() (*restclient.Config, e
 		// TODO: this shouldn't be a global - the client config rules should be
 		//   handling this.
 		defaultConfig, defErr := DefaultClientConfig.ClientConfig()
-		if IsConfigurationInvalid(defErr) && !IsEmptyConfig(err) {
+		if defErr != nil && !IsConfigurationInvalid(defErr) {
 			return mergedConfig, nil
```

Comment:
Partially because we didn't used to hide the error and I didn't want to start hiding it in this set of changes.  I'm almost convinced it would be safe (a default config that can't be loaded can be safely ignored), but not 100%.

It's not technically correct to check DefaultClientConfig here as noted in the TODO.

If mergedClientConfig.ClientConfig() returned a `config, bool, err` where bool was 'used defaults' this code could be drastically simplified.